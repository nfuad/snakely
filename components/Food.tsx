import * as React from 'react'

export default ({ dot }) => (
  <div className="food">
    <style jsx global>
      {`
        .food {
          position: absolute;
          width: 2%;
          height: 2%;
          background: tomato;
          border-radius: 30%;
          border: 1px solid #fff;
          z-index: 1;
          left: ${dot[0]}%;
          top: ${dot[1]}%;
        }
      `}
    </style>
  </div>
)
