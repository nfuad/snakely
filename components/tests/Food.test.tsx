import * as React from 'react'
import { shallow, mount } from 'enzyme'

import Food from '../Food'

const mockData = {
  dot: [0, 1]
}

describe('Food component', () => {
  it('renders', () => {
    shallow(<Food dot={mockData.dot} />)
  })

  it('matches the snapshot', () => {
    const component = mount(<Food dot={mockData.dot} />)
    expect(component).toMatchSnapshot()
  })

  it('updates via props', () => {
    const component = mount(<Food dot={mockData.dot} />)
    expect(component).toMatchSnapshot()

    component.setProps({
      dot: [1, 0]
    })

    expect(component).toMatchSnapshot()
  })
})
