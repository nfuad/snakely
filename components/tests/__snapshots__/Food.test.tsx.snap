// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Food component matches the snapshot 1`] = `
<_default
  dot={
    Array [
      0,
      1,
    ]
  }
>
  <div
    className="jsx-2133318913 food"
  >
    <JSXStyle
      dynamic={
        Array [
          0,
          1,
        ]
      }
      id="1138619144"
    />
  </div>
</_default>
`;

exports[`Food component updates via props 1`] = `
<_default
  dot={
    Array [
      0,
      1,
    ]
  }
>
  <div
    className="jsx-2133318913 food"
  >
    <JSXStyle
      dynamic={
        Array [
          0,
          1,
        ]
      }
      id="1138619144"
    />
  </div>
</_default>
`;

exports[`Food component updates via props 2`] = `
<_default
  dot={
    Array [
      1,
      0,
    ]
  }
>
  <div
    className="jsx-2981255425 food"
  >
    <JSXStyle
      dynamic={
        Array [
          1,
          0,
        ]
      }
      id="1138619144"
    />
  </div>
</_default>
`;
